### PR TITLE
add chapter_sorting by dart

### PR DESCRIPTION
**Pull request by what-is-me**
If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [ ] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [ ] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [ ] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

bubble_sort.dart
```
冒泡排序完成后 nums = [1, 1, 2, 3, 4, 5]
冒泡排序完成后 nums1 = [1, 1, 2, 3, 4, 5]
```
insertion_sort.dart
```
插入排序完成后 nums = [1, 1, 2, 3, 4, 5]
```
merge_sort.dart
```
归并排序完成后 nums = [0, 1, 2, 3, 4, 5, 6, 7]
```
quick_sort.dart
```
快速排序完成后 nums = [0, 1, 2, 3, 4, 5]
快速排序（中位基准数优化）完成后 nums1 = [0, 1, 2, 3, 4, 5]
快速排序（尾递归优化）完成后 nums2 = [0, 1, 2, 3, 4, 5]
```
radix_sort.dart
```
基数排序完成后 nums = [3, 4, 12, 23, 192, 788]
```

Seems you haven't add dart into the doc so I only translate the code.

### CODE REVIEW

**Comment by what-is-me** [comment]
Moreover, minimum value of int wasn't defined in the dart:core, so I had to use -9223372036854775808.
If you are dissatisfied to it, just delete the file `radix_sort.dart`

File: codes/dart/chapter_sorting/bubble_sort.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        nums[j] = nums[j + 1];
+        nums[j + 1] = tmp;
+        flag = true; // 记录交换元素
+      }
+    }
+    if (!flag) break; // 此轮冒泡未交换任何元素，直接跳出
+  }
+}
+
+void main() {
```
Comment (krahets) [inline codes/dart/chapter_sorting/bubble_sort.dart:42]: Add `/* Driver Code */` above. Same as others.

**Review by krahets — COMMENTED** [review]
Thanks for the clean code! Please address the comment.

File: codes/dart/chapter_sorting/radix_sort.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    bucket[d]--; // 将 d 的数量减 1
+  }
+  // 将 tmp 复制到 nums
+  for (int i = 0; i < n; i++) nums[i] = tmp[i];
+}
+
+/* 基数排序 */
+void radixSort(List<int> nums) {
+  // 获取数组的最大元素，用于判断最大位数
+  int ma = -9223372036854775808;
```
Comment (krahets) [inline codes/dart/chapter_sorting/radix_sort.dart:42]: I think the min value of `int` is -2147483648

Could you write it with the binary operator? Like `1 << 31`, which is convenient for remembering it.

Comment (what-is-me) [reply codes/dart/chapter_sorting/radix_sort.dart:42]: emmm, I tested it, in dart, it is the min value of `int`. `int` in dart seems tobe int64 in other language. 
But for unify, I agree with you that we should use 1 << 31.

**Comment by krahets** [comment]
> I fixed the code style and the bug I mentioned [here](https://github.com/krahets/hello-algo/commit/e60dc5dc49bd00e70c3ce7d49b4bcc317bc9cc3f#r100569255)

Thanks!

File: codes/dart/chapter_sorting/radix_sort.dart (hunk 2 above)
Comment (what-is-me) [reply codes/dart/chapter_sorting/radix_sort.dart]: It was mentioned in [int class](https://api.dart.cn/stable/2.19.2/dart-core/int-class.html)

**Review by krahets — APPROVED** [review]
Thanks for the great work!

**Pull request merged**